_posts/2023-01-24-baking-2d-lighting.markdown

```diff
@@ -0,0 +1,187 @@
+
+Unity has many light baking capabilities that helps create Ambient Occlusion and Global Illumination look realistic. Since Void Sols is a top-down game, I knew I didn't need to worry about GI, and could focus solely on figuring out how to get the 2D light sources to bake onto the ground textures.
+
+# The Baking Camera
+
+The metaphor I used to help my conceptualize this task was the idea of "Painting Light". I even ended up naming the script that does most of the work `PaintedLightBG.cs` when I started prototyping and haven't renamed it since. The first thing I needed was some way to take the static light sources the player has lit and "Paint" their light onto a temporary texture. The easiest method that came to mind was to use a literal camera to "take that picture".
+
+So I created a new orthographic camera and positioned it at the center of my current level such that the camera could see the entirety of the level. I immediately ran into a limitation: anywhere beyond the bounds of this camera would not bake lighting. My solution: I'll ignore that problem and cross that bridge when I get there. Moving on, I set the culling mask to only render the layers that should be affected by static light sources. Another limitation: enemies won't cast shadows on baked lighting backgrounds. Early tests showed that this was not a major issue so once again, I moved on.
+
+>![Light Settings](/assets/img/baking-2d-lighting/torch-settings.png "Light Settings") <br>
+>*I set the torches to be off by default and ensured they are all on the same layer. This layer is ignored by the default camera that renders the game, and is only seen by the "Baking Camera" as seen in the image below.*
+
+>![Camera Layers](/assets/img/baking-2d-lighting/baking-camera-layers.png "Baking Camera Layers") <br>
+>*The static walls layer contains parts of the level geometry that I know won't move. The "Baking Camera Only" layer contains only the background layer that the light is baked onto.*
+
+>![Baking Camera](/assets/img/baking-2d-lighting/baking-camera-preview.png "Baking Camera Preview") <br>
+>*Now I created a large, plain, white sprite to fit the entire view of the baking camera. This sprite uses the default "Lit Sprite" material, and will serve as my "canvas" onto which the light will be painted.*
+
+After setting things up in Unity, I could start coding the system that will actually save this painted data to a texture. This uses Render Textures, and I decided to create them at runtime so I could reliably set them up only when I needed them. I also realized I would need two render textures since one would be the direct target of the camera, and the other would be the persistent texture that is continually being painted on.
```

Comment:
"Render Texture" can be treated as a proper noun or not, but capitalize it consistently throughout the page.